src/utils/getWeatherStations.ts

```diff
@@ -23,20 +23,33 @@ export const adjustStationDataset = (dataset: IWeatherStation[]) => {
 
     if (maxDate) {
       const refDate = maxDate.subtract(1, "week");
-      dataset.forEach(function (station) {
+      return datasetArr.map(station => {
         let d = dayjs(station.maxdate);
         if (d && d.isAfter(refDate)) {
-          station.maxdate = "present";
+          return {...station, maxdate: "present"};
+        } else {
+          return station;
         }
       });
     }
```

Comment:
Maybe add a comment about why this is commented out or remove the commented code?